driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -378,6 +406,27 @@ private boolean addConnectionIfUnderMaximum() {
         }
     }
 
+    private PooledConnection tryResurrectFromTrash() {
+        long highestMaxIdleTime = System.currentTimeMillis();
+        PooledConnection chosen = null;
+
+        while (true) {
+            for (PooledConnection connection : trash)
+                if (connection.maxIdleTime > highestMaxIdleTime && connection.maxAvailableStreams() > MIN_AVAILABLE_STREAMS) {
```

Comment:
I think with the new algorithm we can reasonably assume that this will not happen: we only trash when the remaining capacity handles the current load, and inFlight of trashed connections will quickly decrease after that.